IBM/ibm-block-csi-operator
# Bug/csi 3038 node pods in status creating due to bad secrets corelati…

…on (#167)

* changed secrets problem handle

* get the node pods

* add watch on service account

* remove status update

* remove logger from updateStatus

* add logger info on status update

* get when the operator sync the node pods

* add 40 seconds wait after restart

* remove updating IBMBlockCSI status

* add ds_restarted_key variable

* add ds_restarted_key to node SyncFn

* fix wrong variables

* remove declared restarted variables

* remove node syncer logger

* add handle to controller ImagePullBackOff

* answer some PR comments

* change some functions names

* add daemonSet_restarted parameters to verify annotations function

* add areAllPodImagesSynced check in the controller

* updated GetAnnotations function

* remove logger update outside a if

* changed the controller restart handle

* fix more code review comments

* add a name for the serviceaccount return value

* add errors.IsNotFound to getting the controller pod

* add Ctrl+Alt+L

* add CSI-3071 comment

* change the controller restart handle

* cahnge the logger type in the functions

* use logr.Logger in functions

* remove instance from restartControllerPodfromStatefulSet function
